clippy_lints/src/casts/cast_lossless.rs

```diff
@@ -96,7 +96,16 @@ fn should_lint(cx: &LateContext<'_>, cast_from: Ty<'_>, cast_to: Ty<'_>, msrv: M
             };
             !is_isize_or_usize(cast_from) && from_nbits < to_nbits
         },
-        (false, true) if matches!(cast_from.kind(), ty::Bool) && msrv.meets(cx, msrvs::FROM_BOOL) => true,
+        (false, true)
+            if (*cast_from.kind() == ty::Char
+                && matches!(
+                    cast_to.kind(),
+                    ty::Uint(ty::UintTy::U32 | ty::UintTy::U64 | ty::UintTy::U128)
+                ))
+                || (*cast_from.kind() == ty::Bool && msrv.meets(cx, msrvs::FROM_BOOL)) =>
+        {
+            true
+        },
         (_, _) => {
```

Comment:
Would this be easier to read?

```suggestion
        (false, true) => {
            match cast_from.kind() {
                ty::Char => matches!(cast_to.kind(), ty::Uint(ty::UintTy::U32 | ty::UintTy::U64 | ty::UintTy::U128)),
                ty::Bool => msrv.meets(cx, msrvs::FROM_BOOL),
                _ => false,
            }
        },
        (false, false) => {
```

Thanks for working on this so fast!

Reply:
You can make this even more compact with the following code, which even accounts for conversions to `usize`/`isize`, and conversions from `f16` or into `f128` if those are feature enabled:

```rust
    match (cast_from.kind(), cast_to.kind()) {
        (ty::Bool, ty::Uint(_) | ty::Int(_)) => msrv.meets(cx, msrvs::FROM_BOOL),
        (ty::Uint(_), ty::Uint(UintTy::Usize)) => utils::int_ty_to_nbits(cast_from, cx.tcx) <= 16,
        (ty::Uint(UintTy::U8) | ty::Int(_), ty::Int(IntTy::Isize)) => utils::int_ty_to_nbits(cast_from, cx.tcx) <= 16,
        (ty::Uint(UintTy::Usize) | ty::Int(IntTy::Isize), _) | (_, ty::Uint(UintTy::Usize) | ty::Int(IntTy::Isize)) => {
            false
        },
        (ty::Uint(_) | ty::Int(_), ty::Int(_)) | (ty::Uint(_), ty::Uint(_)) => {
            utils::int_ty_to_nbits(cast_from, cx.tcx) < utils::int_ty_to_nbits(cast_to, cx.tcx)
        },
        (ty::Uint(_) | ty::Int(_), ty::Float(fl)) => utils::int_ty_to_nbits(cast_from, cx.tcx) < fl.bit_width(),
        (ty::Char, ty::Uint(_)) => utils::int_ty_to_nbits(cast_to, cx.tcx) >= 32,
        (ty::Float(fl_from), ty::Float(fl_to)) => fl_from.bit_width() < fl_to.bit_width(),
        _ => false,
    }
```

Reply:
Thanks! I made some adjustments as needed.